sound/soc/intel/boards/sof_sdw_maxim.c

```diff
@@ -140,9 +156,19 @@ int sof_sdw_mx8373_init(struct snd_soc_card *card,
 	if (info->amp_num == 2)
 		dai_links->init = spk_init;
 
-	info->codec_card_late_probe = mx8373_sdw_late_probe;
-
-	dai_links->ops = &max_98373_sdw_ops;
+	maxim_part_id = info->part_id;
+	switch (maxim_part_id) {
+	case SOF_SDW_PART_ID_MAX98363:
+		dai_links->ops = &max_98363_sdw_ops;
+		break;
+	case SOF_SDW_PART_ID_MAX98373:
+		info->codec_card_late_probe = mx8373_sdw_late_probe;
+		dai_links->ops = &max_98373_sdw_ops;
+		break;
+	default:
+		dev_err(card->dev, "Invalid maxim_part_id %#x\n", maxim_part_id);
+		return -EINVAL;
+	}
```

Comment:
First add a rename and a code change to support Max98373, then add the support for 98363.

You only did half of what I asked previously.

Reply:
Done

Reply:
Modified the changes to
Commit 1 to change the name and update the makefile
Commit 2 modify and generalize the function, structure names and card components
Commit 3 Updated max98363 changes